src/main/java/com/cloudbees/jenkins/support/SupportCommand.java

```diff
@@ -107,7 +98,7 @@ protected int run() throws Exception {
         }
 
         public OutputStream call() throws IOException {
-            File f = File.createTempFile(filename, ".zip");
+            File f = File.createTempFile(filename, "");
             System.err.println("Creating: " + f);
             return new RemoteOutputStream(new FileOutputStream(f));
         }
```

Comment:
BTW not introduced in this PR, but while you are here please delete the bogus `checkRoles` implementation and extend `MasterToSlaveCallable`.